src/stores/LifecycleStore.ts

```diff
@@ -70,6 +83,40 @@ class LifecycleStore extends AsyncStore<IState> {
     }
 }
 
+async function checkServerVersions(): Promise<void> {
+    const client = MatrixClientPeg.get();
+    if (!client) return;
+    for (const version of SUPPORTED_MATRIX_VERSIONS) {
+        // Check if the server supports this spec version. (`isVersionSupported` caches the response, so this loop will
+        // only make a single HTTP request).
+        // Note that although we do this on a reconnect, we cache the server's versions in memory
+        // indefinitely, so it will only ever trigger the toast on the first connection after a fresh
+        // restart of the client.
+        if (await client.isVersionSupported(version)) {
```

Comment:
It's rather odd that we (sometimes) make another `/versions` request, rather than using the result of the `/versions` request that the client has just successfully done, which caused the transition to the `Syncing` state.

I appreciate that changing that would mostly mean changes in the js-sdk (in the implementation of `isVersionSupported`, I guess), so I'm not asking you to change this now as part of a quick fix to this issue, but maybe you could record the situation in a comment?

Reply:
Yeah, the js-sdk impl also laments the lack of cache expiration which seems roughly in the same category of problems. I can put a comment there?

Reply:
Actually I think it does always re-use the result of the /versions request since it caches the response and of course, in this case, we know it's already done one so it never actually does the request. Noted in 8616cc2e1d.

Reply:
> of course, in this case, we know it's already done one

Per https://github.com/matrix-org/matrix-react-sdk/pull/12280/files/946636614e83b032859f8626e0bfde75ca41df89..8616cc2e1d55cb5bdbf1a5acb0d459e2ffcb994c#r1500777247, I don't think that's right.